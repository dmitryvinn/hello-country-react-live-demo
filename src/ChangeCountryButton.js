import React from "react";

const buttonStyle = {
    border: 'none',
    color: '#ffffff',
    padding: '15px 32px',
    textAlign: 'center',
    textDecoration: 'none',
    display: 'inlineBlock',
    fontSize: '17px',
    backgroundColor: '#555555'
};

class ChangeCountryButton extends React.Component {
    render() {
        return <button className="changeCountry"
                       style={buttonStyle}
                       onClick={this.props.action}>Switch  Country</button>
    }
}

export default ChangeCountryButton;
