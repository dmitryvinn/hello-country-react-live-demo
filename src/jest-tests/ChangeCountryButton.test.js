import React from 'react';
import renderer from 'react-test-renderer';
import ChangeCountryButton from "../ChangeCountryButton";

test('Country Change Button ', () => {
    // Initialize Component
    let component;
    component = renderer.create
    (
        <ChangeCountryButton/>
    );
    // Format result
    let tree = component.toJSON();
    // Assert
    expect(tree).toMatchSnapshot();
});