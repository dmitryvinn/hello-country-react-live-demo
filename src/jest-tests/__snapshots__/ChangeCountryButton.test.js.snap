// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Change Button  1`] = `
<button
  className="changeCountry"
  style={
    Object {
      "backgroundColor": "#555555",
      "border": "none",
      "color": "#ffffff",
      "display": "inlineBlock",
      "fontSize": "17px",
      "padding": "15px 32px",
      "textAlign": "center",
      "textDecoration": "none",
    }
  }
>
  Switch  Country
</button>
`;
